src/irmin/tree.ml

```diff
@@ -266,6 +271,17 @@ module Make (P : S.PRIVATE) = struct
           | Value v -> Lwt.return_some v
           | Hash (repo, k) -> value_of_hash t repo k )
 
+    let to_value_exn t =
```

Comment:
I'm not sure if it's worth implementing `to_value` in terms of `to_value_exn` or vice-versa.